### DISCUSSION TITLE
Lychee does not create the database.

### DISCUSSION THREAD

**Discussion by Ewgeniy77** [LycheeOrg/Lychee#2511] [Q&A] [answered]
Hey, everybody. I have this problem, all the data is correct. I have done this hundreds of times with other CMS. What am I doing wrong? Although I'm doing everything as usual. )

![2024-07-28_22-10-58](https://github.com/user-attachments/assets/4afba588-dd0c-4779-b943-312e9e8c768a)

**Comment by ildyria** [comment]
I have a strong bet that this is related: https://github.com/LycheeOrg/Lychee/issues/1248.

And to clarify, yes Lychee does not create the database, you have to create it yourself with the associated user with proper permissions. Then Lychee will use it.

That being said, we do support other types of DB such as sqlite which then can be created on the fly.

**Reply by Ewgeniy77** [reply]
Thank you for your reply. The empty database has been pre-created, it exists, it doesn't need to be created on the fly.  Lychee cannot connect to an existing database.
The database password does not contain any special characters.

**Reply by ildyria** [reply]
:thinking:

**Reply by ildyria** [reply]
That would be surprising but missing php-mysql dependency ? :(

**Reply by ildyria** [reply]
You are missing the port though... 3306

**Reply by Ewgeniy77** [reply]
the port was specified and not specified, the result is the same. in more than 20 years of amateur work with CMS, this is the first time I see the need to specify a port. ))

I read somewhere that you need to import the database from the file that comes with the kit, but it is not mysql, I tried to import it, the database remained empty.

**Reply by Ewgeniy77** [reply]
> You are missing the port though... 3306

No hoster I've ever dealt with ever specifies a port. Only:
1. Server for site connection: (localhost)
2. User name: (Matches DB name)
3. Databases name: o91510pi_*******

If you want to connect to the database from the outside, then “Server for external connections” is specified instead of “Server for site connection”.

The hoster does not specify any ports anywhere, and everything has always worked fine with other PHP scripts.

![2024-07-29_12-11-53](https://github.com/user-attachments/assets/a207f6e2-b9b2-43d7-8198-0afad59ce7d2)

https://beget.com/en/kb/manual/mysql

**Reply by ildyria** [reply]
That is very very weird. :/

**Reply by ildyria** [reply]
> I read somewhere that you need to import the database from the file that comes with the kit, but it is not mysql, I tried to import it, the database remained empty.

Nah that is not needed.

I would suggest in your case, to switch to sqlite. 
```env
DB_CONNECTION=sqlite
#DB_HOST=
#DB_PORT=
#DB_DATABASE=
#DB_USERNAME=
#DB_PASSWORD=
```

So you can get a feel whether you will like Lychee or not.

**Reply by Ewgeniy77** [reply]
Does this mean that Lychee doesn't support mysql?

**Reply by Ewgeniy77** [reply]
Yeah, so you don't have a direct line. ) Only through PHP extensions. )

You will also need one of these PHP extensions:
SQLite3 for SQLite3 databases
MySQLi (or PDO_MySQL) for MySQL or MariaDB databases
PgSql (or PDO_PGSQL) for PostgreSQL databases

**Reply by ildyria** [reply]
Lychee support all three: MySQL, sqlite and pgsql.
We have cicd running making sure this works.

Could it be that Lychee is not able to read the content of your .env? 

What exactly are you doing in that step? What do you see next ? 

The red writing on top is just a warning. It is not an error.

**Reply by Ewgeniy77** [reply]
I didn't go any further. ) From previous experience, nothing worked without a database connection.
I'll try again the other day. Thank you very much for the replies.

**Reply by ildyria** [reply]
In theory, at the bottom there is a save button and then next

Then you have to wait because it is doing the database initialization.
And then you should see a admin creation panel.

**Comment by Ewgeniy77** [comment]
Hello. I found the time and tried to install Lychee again. I did not create a database, left the default entry:
DB_CONNECTION=sqlite
DB_HOST=
DB_PORT=
*DB_DATABASE=

I ignored the red message that it was not possible to create a database. The installation was successful. But I can not enter the admin panel. I click on the login, but nothing happens.
![2024-08-30_21-03-14](https://github.com/user-attachments/assets/3334f820-ea75-4024-8ab7-2db4461f85d4)

**Reply by ildyria** [reply]
1. Can you check in your console (F12) what is going on ?
2. If you set `LIVEWIRE_ENABLED=false` in your `.env` it will be working.

**Reply by Ewgeniy77** [reply]
Thank you very much for your answer. Yes, the admin login works now. But why do this? ))
And why, when I access the site, do I get this message and a request to go to the folder "https://gallery.lformat.ru/public/"?
![2024-08-30_21-28-40](https://github.com/user-attachments/assets/6f5ddb81-29db-4fe4-9245-aaddf241bc72)

**Reply by ildyria** [reply]
That is because you are not supposed to expose that folder. Your domain name gallery.lformat.ru should be directly pointing to the content of public instead.

**Reply by Ewgeniy77** [reply]
I am completely confused and do not understand where which folders should be. I placed the contents of the "public" folder in the "../gallery.lformat.ru/public_html" folder and all other contents in the root of the site "../gallery.lformat.ru/" but received a message during installation that the folders "/public/uploads", "/public/sym", "/public/dist" were not found. In your documentation, I still have not found a clear answer as to where which folders should be located.

**Reply by Ewgeniy77** [reply]
What folders and files should be in "root" and what in "public_html"?

**Reply by ildyria** [reply]
Okay so you don't have control to where your domain name points ?

**Reply by Ewgeniy77** [reply]
![2024-08-30_22-13-42](https://github.com/user-attachments/assets/933edbad-135d-4d02-ba38-5acb1801dd01)
What folders and files should be in "root" and what in "public_html"?

**Reply by Ewgeniy77** [reply]
> Okay so you don't have control to where your domain name points ?

I am in complete control of everything.

**Reply by ildyria** [reply]
Everything should be in root.
Remove public_html and make your domain name to point to public instead.
_public_ is basically _public_html_. But you can't change the name because it will break lychee.

**Reply by Ewgeniy77** [reply]
I installed a similar CMS, where one part was placed in "root", and the part accessible to the browser was placed in "public_html", and everything worked fine. Maybe you should do the same? It will be much easier. At the moment, your installation is a brain break for ordinary people. That's why so few people know about Lychee.

**Reply by Ewgeniy77** [reply]
Here's what my hosting provider replied:
"There is no way to change DOCUMENT_ROOT as a web server setting. Let's assume that the public part is in the public folder. Then this is implemented as follows (in brackets are the commands for SSH):

Load the site not inside public_html, but next to it
Delete the public_html directory (rm -rf public_html)
Create a symbolic link named public_html pointing to the public directory (ln -s public public_html)"

**Reply by ildyria** [reply]
I will be Dutch and tell you what I think.

First, Lychee is based on Laravel which is pretty much one of the most advanced and stable framework of Php at this time (with Symfony). Laravel uses the convention that the domain name should point to `public`. I am not going to edit the framework when its standard has been well defined over the last 6 years in the case of Lychee.

> That's why so few people know about Lychee.

I disagree. Just because you have to put your hands a bit in the dirt to figure out how to install it does not mean that few people know about Lychee. Furthermore, our software is mainly targeting docker deployment: we have more than 3 millions pulls in comparison.

The biggest problem is that I am the sole dev to maintain Lychee, and as a result I do not have the man power to provide all the shiny features that some of our competitors may provide (on the other hand, we are bleeding edge in term of software quality).
I did try to recruit people and get support, but people do not contribute to project they do not use. So if any other php dev and photographer are willing to join the team, they will be welcomed with wide arms open.

And finally, I am still maintaining Lychee **because I actively use it**, we have only 2 sponsors on GitHub for whom I am really grateful. But aside from that, I own nothing to no-one. Such is the sad state of Open Source Software. Lychee is free, it does not have a payment model and I am not making any money out of it.

**Reply by ildyria** [reply]
> Load the site not inside public_html, but next to it
> Delete the public_html directory (rm -rf public_html)
> Create a symbolic link named public_html pointing to the public directory (ln -s public public_html)"

you can try that indeed.

You may also want to try that method: https://medium.com/@pixelvars/laravel-change-public-html-to-public-html-the-right-way-496431aed94

**Reply by Ewgeniy77** [reply]
> I will be Dutch and tell you what I think.
> 
> First, Lychee is based on Laravel which is pretty much one of the most advanced and stable framework of Php at this time (with Symfony). Laravel uses the convention that the domain name should point to `public`. I am not going to edit the framework when its standard has been well defined over the last 6 years in the case of Lychee.
> 
> > That's why so few people know about Lychee.
> 
> I disagree. Just because you have to put your hands a bit in the dirt to figure out how to install it does not mean that few people know about Lychee. Furthermore, our software is mainly targeting docker deployment: we have more than 3 millions pulls in comparison.
> 
> The biggest problem is that I am the sole dev to maintain Lychee, and as a result I do not have the man power to provide all the shiny features that some of our competitors may provide (on the other hand, we are bleeding edge in term of software quality). I did try to recruit people and get support, but people do not contribute to project they do not use. So if any other php dev and photographer are willing to join the team, they will be welcomed with wide arms open.
> 
> And finally, I am still maintaining Lychee **because I actively use it**, we have only 2 sponsors on GitHub for whom I am really grateful. But aside from that, I own nothing to no-one. Such is the sad state of Open Source Software. Lychee is free, it does not have a payment model and I am not making any money out of it.

Yes, I installed CMS on Laravel. ) And the user part was loaded into public_html, this is 100% reliable information. And ordinary people do not know what "docker" is.

**Comment by Ewgeniy77** [comment]


**Reply by ildyria** [reply]
> It is not clear why there is always a red inscription that the database was not created, 

Lychee does NOT create a database, it will populate a database, but the creation is not in the scopde.

> So, now for any visitor a window for login is shown, how to remove it

Maybe make one of the albums public?

If there are no public album the only action you can really do is ... login, so why not prompt the login dialog immediately?

> You will write again about the parameter "LIVEWIRE_ENABLED=" 

That is because for some reason yours is mis-configured. Anyway, I don't care about Livewire, see here: https://github.com/LycheeOrg/Lychee/discussions/2485 That is why for this reason it is easier for me to tell people to just disable it rather than trying to debug it, and you can keep it disabled. Admin panel will be accessible whether you have it enabled or not.

**Reply by Ewgeniy77** [reply]
> Maybe make one of the albums public?

Yes, that was it. Sorry.

> Lychee does NOT create a database, it will populate a database, but the creation is not in the scopde.

This is the case with many CMS. I create the database myself and specify its data during installation. But this red inscription is very confusing, as if something is wrong. )

Diagnostics gives the following message:

    Warning: /hom**********************************blic/uploads/ has permissions 0775, but should have 2775
    Warning: /hom**********************************blic/uploads/import has permissions 0775, but should have 2775
    Warning: /hom**********************************blic/sym/ has permissions 0775, but should have 2775
    Warning: /home/o/**************************u/storage/image-jobs/ has permissions 0775, but should have 2775
    Warning: /home/o/**************************u/storage/livewire-tmp/ has permissions 0775, but should have 2775

My host wrote that I set the rights to 2775, but I still have this message. What does this mean? Will I not be able to fully work with Lychee?

**Reply by d7415** [reply]
Well that's not what Lychee is seeing so either they're wrong or are doing something weird.

For general effects see https://en.wikipedia.org/wiki/Setuid#When_set_on_a_directory

Lychee-specific permission requirements are in the [documentation](https://lycheeorg.github.io/docs/installation.html#directory-permissions).

It should work like that, but there may be permission problems depending on your setup.

**Comment by Ewgeniy77** [comment]
Lots of mistakes.
![2024-08-31_00-21-34](https://github.com/user-attachments/assets/4959e82d-cffd-431f-90bc-55403b311751)

**Reply by ildyria** [reply]
No, just trying to access a "picture.php" route which your webhost cannot find. So you end up in the honey pot.
Same for "install/picture.php". I have no clue why those are here.

**Comment by Ewgeniy77** [comment]
Diagnostics
    -------
    Warning: /hom**********************************blic/uploads/ has permissions 0755, but should have 2775
    Warning: /hom**********************************blic/uploads/medium2x has permissions 0755, but should have 2775
    Warning: /hom**********************************blic/uploads/medium2x/c4 has permissions 0755, but should have 2775
    Warning: /hom**********************************blic/uploads/medium2x/c4/94 has permissions 0755, but should have 2775
    Warning: /hom**********************************blic/uploads/medium2x/6e has permissions 0755, but should have 2775
    Warning: 159 more directories with wrong permissions
    Warning: Dropbox import not working. dropbox_key is empty.
    Warning: assert.exception is set to false. Lychee assumes that failing assertions throw proper exceptions.
    Warning: zend.assertions is enabled although Lychee is not in debug mode. Outside of debugging, code generation for assertions is recommended to be disabled for efficiency reasons
    Info: Latest version of PHP is 8.3

**Reply by ildyria** [reply, marked as answer]
Just some settings about the directories in upload, it may prevent the creation of thumbs etc. Hopefully not. Given that you seem to have ssh you can just do `chmod 2775 -R uploads` in your public folder and that should resolve quite a few of those issues.

Se what needs to be done to have zend.assertions disabled.
I would also try to see to get your webhost to use 8.3 instead of 8.2 (because that Info is not here for nothing).

From December 1st, don't expect new versions of Lychee to be working on php 8.2.

**Reply by Ewgeniy77** [reply]
PHP 8.3

    Diagnostics
    -------
    Warning: /hom**********************************blic/uploads/ has permissions 0755, but should have 2775
    Warning: /hom**********************************blic/uploads/medium2x has permissions 0755, but should have 2775
    Warning: /hom**********************************blic/uploads/medium2x/c4 has permissions 0755, but should have 2775
    Warning: /hom**********************************blic/uploads/medium2x/c4/94 has permissions 0755, but should have 2775
    Warning: /hom**********************************blic/uploads/medium2x/6e has permissions 0755, but should have 2775
    Warning: 159 more directories with wrong permissions
    Warning: Dropbox import not working. dropbox_key is empty.

**Reply by Ewgeniy77** [reply]
> Just some settings about the directories in upload, it may prevent the creation of thumbs etc. Hopefully not. Given that you seem to have ssh you can just do `chmod 2775 -R uploads` in your public folder and that should resolve quite a few of those issues.

This is a story about some space. )) I have already spent a huge amount of time. I just need to create an album and upload photos, through the browser, of course. That's the whole point, otherwise it is not necessary.

**Comment by Ewgeniy77** [comment]
When I'm in the account I have a button to expand to full screen. But the user does not have such a button, it is only for the album, but the user has already started viewing and he cannot expand the photo to full screen.
![2024-08-31_00-40-44](https://github.com/user-attachments/assets/5f91b97a-b6cb-4fe0-a2e6-c141b4e8ee4b)
![2024-08-31_00-39-55](https://github.com/user-attachments/assets/f9e5f0b0-a20b-4701-bbab-51bd24baa956)

**Reply by ildyria** [reply]
Yeah that's old bugs. It is already fixed in v5 and v6.
https://lycheeorg.github.io/docs/keyboard.html < you may like that.
There is an help menu, but only on livewire front-end.
I am sorry but you will have to be a bit patient for the v6, I have been working my A** off on it, but life has also other priorities (sports, GF, work, family).

**Comment by Ewgeniy77** [comment]
So far I like everything. There is no description for the albums, no description for the photo. How will they be indexed by search engines, how well?

**Reply by ildyria** [reply]
Description is available on photos.
Description is available on albums (just not visible because v4 layout) but already implemeted and support Markdown.

> How will they be indexed by search engines, how well?

Not supported yet, WIP.

**Reply by Ewgeniy77** [reply]
And how to change this inscription:
"All images on this website are subject to copyright by John Smith © 2019"



> Not supported yet, WIP.

Good. You are doing a great job!

**Reply by Ewgeniy77** [reply]
> And how to change this inscription: "All images on this website are subject to copyright by John Smith © 2019"


I found it.)

**Comment by Ewgeniy77** [comment]
Very good CMS, I liked it more than Piwigo and Zenphoto. But very boring and incomprehensible installation for a simple user, on a regular virtual hosting. I would even say that it is impossible to install it on a virtual hosting without errors, access rights 2775 on a virtual hosting are not possible. VPS hosting is already expensive. ((

Unfortunately I'll have to go back to Piwigo. ((
